src/platform/screen.c

```diff
@@ -38,27 +38,47 @@ static int scale_pixels_to_logical(int pixel_value)
     return pixel_value * 100 / scale_percentage;
 }
 
-#ifdef __ANDROID__
-static void set_scale_for_screen(int pixel_width, int pixel_height)
+static int get_max_scale_percentage(int pixel_width, int pixel_height)
+{
+    int width_scale_pct = pixel_width * 100 / MINIMUM.WIDTH;
+    int height_scale_pct = pixel_height * 100 / MINIMUM.HEIGHT;
+    return SDL_min(width_scale_pct, height_scale_pct);
+}
+
+static void set_scale_percentage(int new_scale, int pixel_width, int pixel_height)
 {
-    float scale = android_get_screen_scale();
-    scale = SDL_min(scale, 5.0f);
-    scale = SDL_max(0.5f, scale);
-    scale_percentage = scale * 100;
+    scale_percentage = calc_bound(new_scale, 50, 500);
+
+    if (!pixel_width || !pixel_height) {
+        return;
+    }
+
     // Assure width is at least 640 and height is at least 480
     float width_reference = scale_pixels_to_logical(pixel_width) / (float) MINIMUM.WIDTH;
     float height_reference = scale_pixels_to_logical(pixel_height) / (float) MINIMUM.HEIGHT;
     float minimum_reference = SDL_min(width_reference, height_reference);
     if (minimum_reference < 1.0f) {
```

Comment:
I think you can use `get_max_scale_percentage` here now, right?

Reply:
Yep, fixed.